lib/internal/fs/watchers.js

```diff
@@ -270,10 +295,98 @@ ObjectDefineProperty(FSEvent.prototype, 'owner', {
   set(v) { return this[owner_symbol] = v; }
 });
 
+async function* watch(filename, options = {}) {
+  const path = toNamespacedPath(getValidatedPath(filename));
+  validateObject(options, 'options');
+
+  const {
+    persistent = true,
+    recursive = false,
+    encoding = 'utf8',
+    signal,
+  } = options;
+
+  validateBoolean(persistent, 'options.persistent');
```

Comment:
The validation logic here is _slightly_ different from the logic in `fs.watch` is that intentional?

Reply:
Yeah, I wanted to make it a bit tighter here.